src/AI/Attack.pm

```diff
@@ -39,11 +39,7 @@ sub process {
 	Benchmark::begin("ai_attack") if DEBUG;
 	my $args = AI::args;
 
-	if (
-		   (AI::action eq "attack" && $args->{ID})
-		|| (AI::action eq "route" && AI::action (1) eq "attack" && $args->{attackID})
-		|| (AI::action eq "move" && AI::action (2) eq "attack" && $args->{attackID})
-	) {
+	if (shouldAttack($action, $args)) {
```

Comment:
`$action` is not defined.
maybe:
`my $action = AI::action;`

Reply:
Sorry, a I forget this -.-. Thanks for you review Alison :D